lib/rtp_h264/nal_formats/fu.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Membrane.Element.RTP.H264.FU do
+  @moduledoc """
+  Module responsible for parsing H264 Fragmentation Unit.
+  """
+  use Bunch
+  alias Membrane.Element.RTP.H264.FU.Header
+  alias Membrane.Element.RTP.H264.{Depayloader, NALHeader}
+
+  defstruct [:last_seq_num, data: []]
+
+  @type t :: %__MODULE__{
+          data: [binary()],
+          last_seq_num: nil | Depayloader.sequence_number()
+        }
+
+  defguardp next?(last_seq_num, next_seq_num) when last_seq_num + 1 == next_seq_num
+
+  @doc """
+  Parses H264 Fragmentation Unit
+
+  If a packet that is being parsed is not considered last then a `{:incomplete, t()}`
+  tuple  will be returned.
+  In case of last packet `{:ok, {type, data}}` tuple will be returned, where data is `NAL Unit`
+  created by concatenating subsequent Fragmentation Units.
+  """
+
+  @spec parse(binary(), Depayloader.sequence_number(), t) ::
+          {:ok, {binary(), NALHeader.type()}}
+          | {:error, :packet_malformed | :invalid_first_packet}
+          | {:incomplete, t()}
+  def parse(data, seq_num, acc) do
+    # actual code
+    data
+    |> Header.parse()
+    ~>> ({:ok, {header, value}} -> do_parse(header, value, seq_num, acc))
+  end
+
+  defp do_parse(header, data, seq_num, acc)
+
+  defp do_parse(%Header{end_bit: true, type: type}, data, seq_num, %__MODULE__{
```

Comment:
@erszcz Found an error here. If end bit is true and last_seq_num is nil guard will try add 1 to nil and crash.